css/css-fonts/generic-family-keywords-002.html

```diff
@@ -0,0 +1,32 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Generic font family: invalid -webkit-generic-family values</title>
+    <link rel="help" href="https://drafts.csswg.org/css-fonts-4/#generic-font-families">
+    <meta name="assert" content="-webkit-generic-family are treated as <family-name> strings, not <generic-family> keywords."/>
+    <link rel="match" href="generic-family-keywords-002-ref.html"/>
+    <link rel="stylesheet" type="text/css" href="/fonts/ahem.css"/>
+    <style>
+      span { color: blue; }
+    </style>
+  </head>
+  <body>
+    <p>This test passes if the blue text after each colon renders as a square.</p>
+    <p>This assumes that there are no local fonts of name
+      <code>-webkit-&lt;generic-family&gt;</code> installed.</p>
+    <p>-webkit-serif: <span style="font-family: -webkit-serif, Ahem;">X</span></p>
+    <p>-webkit-sans-serif: <span style="font-family: -webkit-sans-serif, Ahem;">X</span></p>
```

Comment:
I don’t quite understand the motivation to add these keywords to the web platform. I thought Web Platform Tests was supposed to be testing according to the spec, not prefixed quirks of individual implementations.

Reply:
Unless I'm missing something the css-fonts and cssom specifications don't say anything about treating font-family values with browser-vendor prefix specially, so they are not "keywords" but actually <family-name> values. AFAIK, all these tests conform to the specifications. You might not like them but until I'm proved wrong, the violation of the spec is really in Chromium and WebKit... I hope at least the Chromium project could fix that in the future. Even if WebKit will probably continue to support these, they can just skip the tests and at least I find it useful to make us aware of the issue (e.g. did you really know that "-webkit-serif" was recognized by WebKit? Or serialized as serif?)

Reply:
> did you really know that "-webkit-serif" was recognized by WebKit? Or serialized as serif?

Yes, but I don’t think it’s desirable. I’m surprised that -webkit-serif is something that should be tested by WPT. Do you think it should be in the spec too?

Reply:
I don't think my personal opinion really matters here but since you are asking in general I don't think proprietary "-webkit-*" should be mentioned in the spec or supported by browsers as that's bad for the web (from your last comment, it seems you agree). Anyway this discussion about whether to support browser prefix should be done on the spec side or on the web engine side, not in WPT. If the spec is changed I can update my tests. If a web engine decides to violate the specification they can add specific test expectations in their own repo (this should be even easier now that I converted them to testharness tests).

Anyway as you initially said, what is important for this PR is whether tests added to the WPT repository conform to the specification. Again, reading the spec all these "-webkit-*" are treated as <family-name> not <generic-name> so unless I'm wrong, my tests are perfectly valid here. Do you disagree?

Reply:
So to restate this, Fred's point -- if I'm understanding correctly -- is that the spec (quite properly) says nothing about a value like `-webkit-serif` being a `<generic-name>`. So the spec-compliant interpretation is that it is a `<family-name>` just like `times` or `helvetica` or `MyRandomFamilyName`.

So it's valid to test that the browser treats it as such, and does *not* interpret it as a `<generic-name>`.

Reply:
Oh, that makes much more sense, I’m so sorry for the misunderstanding!!

Re-reading through this thread, you were very clear from the beginning, multiple times, and I was just not getting it. Thank you for your patience! I’m so sorry for all the confusion.

Reply:
No worries, I guess it happens some time... at least it gave me time to improve the tests. thanks for the approval, I'm going to merge it now.